### PR TITLE
Topology test

### PR DESCRIPTION
**Pull request by syclops**
Added the beginnings of a unit testing file for lib/topology.py.
<a href='#crh-start'></a><a href='#crh-data-%7B%22processed%22%3A%20%5B%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-107695496%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-107871115%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-111998747%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-114076507%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23discussion_r32824558%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23discussion_r32831802%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23discussion_r32833332%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23discussion_r32926126%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-114076716%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-114088200%22%2C%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-114093294%22%5D%2C%20%22comments%22%3A%20%7B%22General%20Comment%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-107695496%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22There%27s%20a%20%5C%22master%5C%22%20issue%20%28%23129%29%20where%20people%20pick%20modules%20to%20prepare%20unit%20tests%20for%20them%2C%20please%20check%20it%20out%20first%20next%20time.%5Cr%5CnDo%20you%20want%20to%20continue%20working%20on%20unit%20tests%20for%20%60lib.topology%60%3F%20There%27s%20no%20problem%20with%20that%2C%20as%20I%20haven%27t%20done%20much%20yet%20anyway%3B%20just%20want%20to%20avoid%20overlaps.%22%2C%20%22created_at%22%3A%20%222015-06-01T20%3A21%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1120468%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/rev112%22%7D%7D%2C%20%7B%22body%22%3A%20%22Okay%2C%20sure.%20I%27ve%20updated%20the%20list%20to%20reflect%20this.%5Cn%5Cn-Steve%5Cn%5CnOn%206/1/15%2010%3A21%20PM%2C%20Anton%20Ovchinnikov%20wrote%3A%5Cn%3E%5Cn%3E%20There%27s%20a%20%5C%22master%5C%22%20issue%20%28%23129%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/issues/129%3E%29%20where%20people%20pick%5Cn%3E%20modules%20to%20prepare%20unit%20tests%20for%20them%2C%20please%20check%20it%20out%20first%20next%5Cn%3E%20time.%5Cn%3E%20Do%20you%20want%20to%20continue%20working%20on%20unit%20tests%20for%20%7Clib.topology%7C%3F%5Cn%3E%20There%27s%20no%20problem%20with%20that%2C%20as%20I%20haven%27t%20done%20much%20yet%20anyway%3B%20just%5Cn%3E%20want%20to%20avoid%20overlaps.%5Cn%3E%5Cn%3E%20%5Cu2014%5Cn%3E%20Reply%20to%20this%20email%20directly%20or%20view%20it%20on%20GitHub%5Cn%3E%20%3Chttps%3A//github.com/netsec-ethz/scion/pull/143%23issuecomment-107695496%3E.%5Cn%3E%5Cn%5Cn%22%2C%20%22created_at%22%3A%20%222015-06-02T08%3A50%3A58Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1296929%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/syclops%22%7D%7D%2C%20%7B%22body%22%3A%20%22It%20looks%20like%20this%20depends%20on%20%23142%2C%20so%20i%27ll%20leave%20reviewing%20it%20until%20that%27s%20finalised.%22%2C%20%22created_at%22%3A%20%222015-06-15T09%3A40%3A24Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-none%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-06-22T11%3A48%3A51Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Add%20a%20comment%20here%20when%20%23142%20is%20merged%2C%20and%20i%27ll%20review%20this%20properly.%22%2C%20%22created_at%22%3A%20%222015-06-22T11%3A50%3A23Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Well%20fuck.%20This%20branch%20wasn%27t%20up%20to%20date%2C%20and%20has%20reverted%20a%20Lot%20of%20stuff%20on%20master%20%3A/%22%2C%20%22created_at%22%3A%20%222015-06-22T12%3A42%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Something%20is%20wrong%20here%2C%20i%27ve%20contacted%20github%20support%20to%20figure%20out%20what%20happened.%22%2C%20%22created_at%22%3A%20%222015-06-22T12%3A59%3A47Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22General%20Comment%22%7D%2C%20%22Pull%20412b35a89cc676c1878a83284cb24b50426bbb3d%20test/topology_test.py%201%22%3A%20%7B%22html_url%22%3A%20%22https%3A//github.com/netsec-ethz/scion/pull/143%23discussion_r32824558%22%2C%20%22comments%22%3A%20%5B%7B%22body%22%3A%20%22Please%20rename%20to%20%60lib_topology_test.py%60%22%2C%20%22created_at%22%3A%20%222015-06-19T12%3A36%3A46Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22Fixed.%20If%20you%20have%20no%20other%20comments%2C%20please%20merge%20this.%22%2C%20%22created_at%22%3A%20%222015-06-19T14%3A08%3A49Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/1296929%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/syclops%22%7D%7D%2C%20%7B%22body%22%3A%20%22This%20still%20depends%20a%20lot%20on%20%23142.%20Lets%20get%20that%20sorted%20first%22%2C%20%22created_at%22%3A%20%222015-06-19T14%3A24%3A12Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%2C%20%7B%22body%22%3A%20%22%3A%2B1%3A%3Ca%20href%3D%27%23crh-update-complete%27%3E%3C/a%3E%22%2C%20%22created_at%22%3A%20%222015-06-22T11%3A48%3A52Z%22%2C%20%22user%22%3A%20%7B%22avatar_url%22%3A%20%22https%3A//avatars.githubusercontent.com/u/6919822%3Fv%3D3%22%2C%20%22html_url%22%3A%20%22https%3A//github.com/kormat%22%7D%7D%5D%2C%20%22title%22%3A%20%22File%3A%20test/topology_test.py%3AL1-156%22%7D%7D%7D'></a>
<a href='https://www.codereviewhub.com/'></a>
- [ ] <a href='#crh-comment-General Comment'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/143#issuecomment-107695496'>General Comment</a></b>
- <a href='https://github.com/rev112'></a> There's a "master" issue (#129) where people pick modules to prepare unit tests for them, please check it out first next time.
  Do you want to continue working on unit tests for `lib.topology`? There's no problem with that, as I haven't done much yet anyway; just want to avoid overlaps.
- <a href='https://github.com/syclops'></a> Okay, sure. I've updated the list to reflect this.
  -Steve
  On 6/1/15 10:21 PM, Anton Ovchinnikov wrote:
  >
  > There's a "master" issue (#129
  > https://github.com/netsec-ethz/scion/issues/129) where people pick
  > modules to prepare unit tests for them, please check it out first next
  > time.
  > Do you want to continue working on unit tests for |lib.topology|?
  > There's no problem with that, as I haven't done much yet anyway; just
  > want to avoid overlaps.
  >
  > —
  > Reply to this email directly or view it on GitHub
  > https://github.com/netsec-ethz/scion/pull/143#issuecomment-107695496.
  >
- <a href='https://github.com/kormat'></a> It looks like this depends on #142, so i'll leave reviewing it until that's finalised.
- <a href='https://github.com/kormat'></a> :+1:<a href='#crh-update-none'></a>
- <a href='https://github.com/kormat'></a> Add a comment here when #142 is merged, and i'll review this properly.
- <a href='https://github.com/kormat'></a> Well fuck. This branch wasn't up to date, and has reverted a Lot of stuff on master :/
- <a href='https://github.com/kormat'></a> Something is wrong here, i've contacted github support to figure out what happened.
- [x] <a href='#crh-comment-Pull 412b35a89cc676c1878a83284cb24b50426bbb3d test/topology_test.py 1'></a> &nbsp;<b><a href='https://github.com/netsec-ethz/scion/pull/143#discussion_r32824558'>File: test/topology_test.py:L1-156</a></b>
- <a href='https://github.com/kormat'></a> Please rename to `lib_topology_test.py`
- <a href='https://github.com/syclops'></a> Fixed. If you have no other comments, please merge this.
- <a href='https://github.com/kormat'></a> This still depends a lot on #142. Lets get that sorted first

<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/143?mark_as_completed=1'></a>&nbsp;<a href='https://www.codereviewhub.com/netsec-ethz/scion/pull/143?approve=1'></a>&nbsp;<a href='https://github.com/netsec-ethz/scion/pull/143'></a>
<a href='#crh-end'></a>

### CODE REVIEW

File: test/topology_test.py [hunk 1]
```diff
@@ -0,0 +1,155 @@
+# Copyright 2015 ETH Zurich
```
Comment (kormat) [inline test/topology_test.py:1]: Please rename to `lib_topology_test.py`

Comment (syclops) [inline test/topology_test.py:1]: Fixed. If you have no other comments, please merge this.

Comment (kormat) [inline test/topology_test.py:1]: This still depends a lot on #142. Lets get that sorted first

Comment (kormat) [inline test/topology_test.py:1]: :+1:<a href='#crh-update-complete'></a>

**Pull request merged**